### PR TITLE
feat: add HashMap.HashMap.Entry utility type

### PR DESCRIPTION
**Pull request by vinassefranche**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds HashMap.HashMap.Entry utility type which is just `[HashMap.HashMap.Key<>, HashMap.HashMap.Value<>]`

## Related

- Related Issue #
- Closes #

### CODE REVIEW

**Comment by vinassefranche** [comment]
I tried to add some types tests with:
```
// $ExpectType ["k", "v"]
export type E = HashMap.HashMap.Entry<typeof hmLiterals>
```

but I had this error and did not know what to do about it:
```
ERROR: 31:1  expect  TypeScript@5.5 expected type to be:
  ["k", "v"]
got:
  E
```

Tell me if you have an idea on how to fix it

**Comment by gcanti** [comment]
you can use
```ts
import { hole, pipe } from "effect/Function"
import type * as Types from "effect/Types"

// $ExpectType ["k", "v"]
hole<Types.Simplify<HashMap.HashMap.Entry<typeof hmLiterals>>>()
```

**Comment by vinassefranche** [comment]
> you can use
> 
> ```ts
> import { hole, pipe } from "effect/Function"
> import type * as Types from "effect/Types"
> 
> // $ExpectType ["k", "v"]
> hole<Types.Simplify<HashMap.HashMap.Entry<typeof hmLiterals>>>()
> ```

Great! Added to the P.R.!

File: packages/effect/src/HashMap.ts [hunk 1]
```diff
@@ -66,6 +66,22 @@ export declare namespace HashMap {
...
+   *
+   * @example
+   * import { HashMap } from "effect"
+   *
+   * declare const hm: HashMap.HashMap<string, number>
+   *
+   * // $ExpectType [string, number]
+   * type V = HashMap.HashMap.Entry<typeof hm>
+   *
+   * @since 2.0.0
```
Comment (gcanti) [inline packages/effect/src/HashMap.ts:81]:
```suggestion
   * @since 3.9.0
```

Comment (vinassefranche) [reply packages/effect/src/HashMap.ts]: Sorry, fixed!

**Pull request merged**